gossip/server.go

```diff
@@ -88,6 +88,10 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 	if err != nil {
 		return nil, util.Errorf("local addr %s could not be converted to net.Addr: %s", args.LAddr, err)
 	}
+	// Verify that the client connection is valid and hasn't been closed.
+	if _, ok := s.lAddrMap[lAddr.String()]; !ok {
```

Comment:
Ah, you're right. I didn't realize that we made that address available on the client side so I just assumed it must be the listening address. But we've broken this with the addition of reconnects to `rpc.Client`: a client no longer has a stable local address, so I think we could see a race with a reconnection occurring between the call to `Client.LocalAddr()` and the call actually being sent.